mesonbuild/backend/ninjabackend.py

```diff
@@ -1185,7 +1185,7 @@ def generate_custom_target(self, target):
             elem.add_item('pool', 'console')
         full_name = Path(target.subdir, target.name).as_posix()
         elem.add_item('COMMAND', cmd)
-        elem.add_item('description', f'Generating {full_name} with a custom command{cmd_type}')
+        elem.add_item('description', target.description.format(full_name) + cmd_type)
```

Comment:
Such customization seems pretty useful to have from public API too. IIRC last time I looked at this, `custom_target('name', ...)` is already only used for ninja's description and can be pretty free-form. Maybe we can change this to `Generating {target.name} in {target.subdir} with a custom command{cmd_type}`.

That way, instead of adding a new description attribute, the target name could be e.g. `'yelp doc'`.

Would have to double check if target's name is used by vs backend in a different way.

Reply:
Alternatively, we could also just deprecate custom_target() name argument because it already defaults to first output path anyway, and add a description kwarg.

Note that this is not blocker, this PR is already a nice improvements as-is.

Reply:
Imho, the `name` attribute should be used to create an alias target.